src/Compilers/Core/CodeAnalysisTest/Diagnostics/SuppressMessageAttributeCompilerTests.cs

```diff
@@ -150,6 +150,14 @@ public class C2
                 diagnostics: new[] { diagnosticC, diagnosticC1, diagnosticC2 });
         }
 
+        [Fact, WorkItem(59746, "https://github.com/dotnet/roslyn/issues/59746")]
+        public async Task SuppressMessageAttributeOnLambda()
+        {
+            await VerifyCSharpAsync(@"using System.Diagnostics.CodeAnalysis;
+
+var lambda = [SuppressMessage("""", ""Comment"")] () => /**/Object.ReferenceEquals(1, 2);", new[] { new WarningOnCommentAnalyzer() });
```

Comment:
We need tests for the rest of the cases covered by the change.

Reply:
Other cases cannot have attributes. I'll add direct semantic model tests.